src/analysis/retail/priest/discipline/constants.ts

```diff
@@ -65,6 +65,26 @@ export const ATONEMENT_SOURCE_FILTER: SpellFilter = [
   SPELLS.ASCENDED_ERUPTION,
 ];
 
+// Spells which are consumed by the holy buff of Twilight Equilibrium
+export const HOLY_DAMAGE_SPELLS = [
+  SPELLS.SMITE.id,
+  SPELLS.PENANCE.id,
+  SPELLS.HALO_DAMAGE.id,
+  TALENTS_PRIEST.PURGE_THE_WICKED_TALENT.id,
+  TALENTS_PRIEST.POWER_WORD_SOLACE_TALENT.id,
+  SPELLS.DIVINE_STAR_DAMAGE.id,
+  TALENTS_PRIEST.HOLY_NOVA_TALENT.id,
+  TALENTS_PRIEST.LIGHTS_WRATH_TALENT.id,
+];
+// Spells which are consumed by the shadow buff of Twilight Equilibrium
+export const SHADOW_DAMAGE_SPELLS = [
+  SPELLS.SHADOW_WORD_PAIN.id,
+  SPELLS.MIND_BLAST.id,
+  TALENTS_PRIEST.MINDGAMES_TALENT.id,
+  TALENTS_PRIEST.SHADOW_WORD_DEATH_TALENT.id,
+  TALENTS_PRIEST.SCHISM_TALENT.id,
+];
```

Comment:
Does TE buff DoTs for their full duration? If so, your current handling will miss most/all of the boosted periodic damage (and probably will also false positive when you have the buff while periodic sources are ticking?)

Reply:
Yup, that's right and good catch! 

I'll amend this probably tomorrow. PTW is permanently buffed once it's procced once by TE. 

I think once the cast triggers it, I will keep a boolean like buffedByTe on dot ticks. Luckily extensions and duplications of the dot all have the same behaviour, so I can just remove the boolean if the dot falls off the main target. What do you think @kfinch

Reply:
I've dealt with something similar in a few of my modules - see the handling of which targets have BT Rip in `Bloodtalons.tsx` - tracked with a set indexed by encodedTargetId called `targetsWithBtRip`. A similar handler should work with you, but triggering only on Apply and not Refresh.